autopilot/erdosrenyi.go

```diff
@@ -0,0 +1,267 @@
+package autopilot
+
+import (
+	"fmt"
+	prand "math/rand"
+	"time"
+
+	"github.com/roasbeef/btcd/btcec"
+	"github.com/roasbeef/btcutil"
+)
+
+// ConstrainedErdosRenyi is an implementation of the AttachmentHeuristic
+// interface that implement a constrained Erdos Renyi heuristic. This means that
+// given a threshold to allocate to automatic channel establishment, the
+// heuristic will attempt to randomly connecting to nodes, selected by sampling
+// from a uniform  distribution.
+type ConstrainedErdosRenyi struct {
```

Comment:
One possible refactoring might be to rename `ConstrainedPrefAttachment` to sth more generic (what's the common denominator of the two heuristics?), and just add a field for `Select` functions, since as far as I can see right now, those are the only funcs that are different?
I think that you even hinted that most of the `Select` code is the same, so you could maybe even find a name for that difference and have a configurable field for it.
Also not sure if that's the preferred way in Go, but would dive into it some more tomorrow.